src/dvc_data/index/fetch.py

```diff
@@ -93,7 +110,11 @@ def fetch(
                     storage="data",
                     jobs=jobs,
                     callback=cb,
+                    onerror=partial(_onerror, data, cache, failed_keys),
```

Comment:
On a long term, I think we should fix `apply` to return the stats instead of misusing `onerror` (I see that this is due to `generic.transfer`).